smartsim/_core/control/previewrenderer.py

```diff
@@ -0,0 +1,116 @@
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+import typing as t
+import jinja2
+import logging
+from ...log import log_to_file_preview
+import smartsim._core._cli.utils as _utils
+from ..._core.config import CONFIG
+from ..._core.control import Manifest
+
+_OutputFormatString = t.Optional[t.Literal["html"]]
+_VerbosityLevelString = t.Literal["info", "debug", "developer"]
+
+def render(
+    experiment: t.Any,
+    manifest: t.Optional[Manifest] = None,
+    verbosity_level: _VerbosityLevelString = "info",
+    output_format: _OutputFormatString = None,
+    output_filename: t.Optional[str] = None,
+) -> str:
+    """
+    Render the template from the supplied entity
+    """
+
+    _check_verbosity_level(verbosity_level)
+
+    loader = jinja2.PackageLoader("templates")
+    env = jinja2.Environment(loader=loader, autoescape=True)
+    if output_format:
+        _check_output_format(output_format)
+        output_filename = _check_output_filename(output_filename)
+
+        tpl = env.get_template(f"base_{output_format}.template")
+        rendered_preview = tpl.render(
+            exp_entity=experiment,
+            manifest=manifest,
+            config=CONFIG,
+            verbosity_level=verbosity_level,
+        )
+        preview_to_file(rendered_preview, output_filename)
+    else:
+        if output_filename:
+            raise ValueError(
+                "Output filename is only a valid parameter when an output \
+format is specified"
+            )
+        tpl = env.get_template("base.template")
+        rendered_preview = tpl.render(
+            exp_entity=experiment,
+            manifest=manifest,
+            config=CONFIG,
+            verbosity_level=verbosity_level,
+        )
+
+
+    return rendered_preview
+
+
+
+def preview_to_file(content: str, file_name: str) -> None:
+    logger = logging.getLogger("preview-logger")
+    log_to_file_preview(filename=file_name, logger=logger)
+    logger.info(content)
+
+
+def _check_output_format(output_format: str) -> None:
+    if not output_format.startswith("html"):
+        raise ValueError("The only valid currently available is html")
+
+
+def _check_output_filename(output_filename: t.Optional[str]) -> str:
+    if not output_filename:
+        raise ValueError("An output filename is required when an output format is set.")
+
+    return output_filename
+
+def _check_verbosity_level(
+    verbosity_level: _VerbosityLevelString
+) -> None:
+    """
+    Check verbosity_level
+    """
+    if verbosity_level == "debug":
+        raise NotImplementedError
+    if verbosity_level == "developer":
+        raise NotImplementedError
+    verbosity_level = t.cast(_VerbosityLevelString, verbosity_level)
+    if (
+        not verbosity_level.startswith("info")
+        and not verbosity_level.startswith("debug")
+        and not verbosity_level.startswith("developer")
```

Comment:
consider `if verbosity_level not in ['info','debug', 'developer']` instead?

Reply:
changed